### PR TITLE
Fix Old Restored Tab Added to Yesterday When Selected

### PR DESCRIPTION
**Pull request by evancrow**
If you deleted a tab that was in the yesterday section and then restored it, it would be selected but still in the yesterday section. The fix was to move the call that updates the `CardGrid` to after the tab selected, not before.

### How was this tested?
- [x] I tested this manually

### Screenshots and screen recordings
Fixed:

https://user-images.githubusercontent.com/37356984/177623393-94ad4281-3353-48d8-9c5e-1b6223c949a9.mp4

### CODE REVIEW

File: Client/Frontend/Browser/Tabs/TabManager/TabManager+AddTab.swift [hunk 1]
```diff
@@ -281,6 +279,7 @@ extension TabManager {
             recentlyClosedTabs.remove(at: index)
         }
 
+        updateAllTabDataAndSendNotifications(notify: true)
```
Comment (darinwf) [inline Client/Frontend/Browser/Tabs/TabManager/TabManager+AddTab.swift:282]: I'm concerned about the call to `selectTab` above that passes `notify: true`. That means some code may see a selected tab that does not appear to exist as part of the existing tabs.

**Comment by darinwf** [comment]
Also, please add a test :)

File: Client/Frontend/Browser/Tabs/TabManager/TabManager+AddTab.swift (hunk 1 above)
Comment (evancrow) [reply Client/Frontend/Browser/Tabs/TabManager/TabManager+AddTab.swift:282]: I can make a change where `updateAllTabDataAndSendNotifications` stays where it originally was, but then we make another call to `updateActiveTabsAndSendNotifications` after selecting the tab. Is that better?

File: Client/Frontend/Browser/Tabs/TabManager/TabManager+AddTab.swift [hunk 2]
```diff
@@ -281,6 +281,7 @@ extension TabManager {
             recentlyClosedTabs.remove(at: index)
         }
 
+        updateActiveTabsAndSendNotifications(notify: true)
```
[the remark this answers is not in the document]
Comment (evancrow) [reply Client/Frontend/Browser/Tabs/TabManager/TabManager+AddTab.swift:284]: Following up on what we discussed in our 1:1, refactored `SelectTab`.

**Comment by evancrow** [comment]
Fixed: #4025

**Review by darinwf — APPROVED** [review]

File: Client/Frontend/Browser/Tabs/TabManager/TabManager.swift [hunk 3]
```diff
@@ -272,7 +272,7 @@ class TabManager: NSObject {
         selectedTab.lastExecutedTime = Date.nowMilliseconds()
         selectedTab.applyTheme()
 
-        // Tab data needs to be updated when a tab is brought back from archives.
+        // Tab data needs to be updated after the lastExecutedTime is modified.
```
Comment (darinwf) [inline Client/Frontend/Browser/Tabs/TabManager/TabManager.swift:275]: perfect, thanks!

**Review by darinwf — APPROVED** [review]

**Pull request merged**